### PR TITLE
[react-navigation] Adds createNavigator, createNavigationContainer

### PR DESCRIPTION
**Pull request by iRoachie**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Adds createNavigator, createNavigationContainer. Url and reference in definitions

### CODE REVIEW

**Comment by iRoachie** [comment]
@huhuanming Can you check this for me>?

**Comment by huhuanming** [comment]
😄
sure... 
let me check that.

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -659,6 +666,31 @@ export function StackRouter(
...
+
+/**
+ * Create Navigator
+ *
+ * @see https://github.com/react-community/react-navigation/blob/master/src/navigators/createNavigator.js
+ */
+export function createNavigator(
+  router: NavigationRouter<any, any, any>,
+  routeConfigs?: NavigationRouteConfigMap,
+  navigatorConfig?: any,
```
Comment (huhuanming) [inline types/react-navigation/index.d.ts]: is the `navigatorConfig` type `StackNavigatorConfig`?

**Review by huhuanming — CHANGES_REQUESTED** [review]

File: types/react-navigation/index.d.ts (hunk 1 above)
Comment (iRoachie) [inline types/react-navigation/index.d.ts]: Not sure actually. The flow typings have ‘any’ there as well.

Comment (iRoachie) [inline types/react-navigation/index.d.ts]: Nah its doesn't have to be, it could be Tab, Drawer, Stack. The flow types changed recently.

Comment (iRoachie) [inline types/react-navigation/index.d.ts]: I'm not sure actually. The flow types changed since I made the PR so I have to update this.

**Comment by mhegazy** [comment]
@iRoachie are you sill interested in pursuing this PR? if so can we resolve the comment above?

**Comment by iRoachie** [comment]
@mhegazy Updated

**Pull request merged**